### DISCUSSION TITLE
How do you get a secret?

### DISCUSSION THREAD

**Discussion by oFinally** [community/community#53300] [New to GitHub]
I'm really new to GitHub, and I'm having some sort of trouble.

I want to know how to create and use secrets on GitHub. I mean, I searched many tutorials but I don't really understand it. Is there any simple way where you can just get the secret with a line of javascript, python, etc. code? (like python's `os.getenv` and those stuff)

**Comment by airtower-luna** [comment]
Using `os.getenv()` (or maybe nicer, `os.environ`) goes into the right direction. The trick is that you'll need to put the secret into an environment variable for the step in which your code runs, e.g.:

```yaml
- run: python your_script.py
  env:
    YOUR_SECRET: ${{ secrets.YOUR_SECRET }}
```

Your script can then access `os.environ['YOUR_SECRET']` to get the value. Of course you don't have to name the environment variable like the secret, it's just an obvious choice. :slightly_smiling_face:

**Reply by oFinally** [reply]
thanks, but what about Javascript? also humour me, but the code doesn't look like python

**Reply by airtower-luna** [reply]
> also humour me, but the code doesn't look like python

That's a snippet of a GitHub Actions workflow (which uses a custom YAML-based syntax), which is where you could use your secrets. My point is that the Python script called in the `run` step would be able to read the secret from the `YOUR_SECRET` environment variable. If you don't use Actions, I wonder how you got to [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) at all. :wink: 

For other languages the principle is that same: Put the secret into the environment, and access it from there. Make sure to expose it only where actually needed.

**Reply by oFinally** [reply]
i read some docs and it says to make a directory called `.github/workflows` and then name it. is there a specific name for it?

also is javascript's getting env `process.env["YOUR_SECRET"]`, and if im using javascript do i set the first line of code to `run: javascript your_script.js`? thanks though.

I'm also getting a bug from your code earlier.

**Reply by airtower-luna** [reply]
> i read some docs and it says to make a directory called `.github/workflows` and then name it. is there a specific name for it?

Workflow files *must* be in `.github/workflows/`, and have `.yaml` or `.yml` as the file ending. And of course must use [valid workflow syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions), the snippet I posted above would belong into the [`steps` section](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idsteps) of a job, after checking out your code and setting up Python.

See also: https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions

> also is javascript's getting env `process.env["YOUR_SECRET"]`, and if im using javascript do i set the first line of code to `run: javascript your_script.js`?

I assume by Javascript you mean Node? I can't really comment on that. A [`run` step](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun) runs the command you give it in a shell, there's no magic for programming languages. So you'd probably run `npm` or `npx` or something.
